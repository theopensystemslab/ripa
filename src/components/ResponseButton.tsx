import Box from "@material-ui/core/Box";
import ButtonBase from "@material-ui/core/ButtonBase";
import { makeStyles } from "@material-ui/core/styles";
import classNames from "classnames";
import React from "react";

const useStyles = makeStyles(theme => ({
  root: {
    border: 0,
    color: "currentColor",
    textAlign: "left",
    display: "flex",
    alignItems: "flex-start",
    justifyContent: "flex-start",
    width: "100%",
    position: "relative",
    fontSize: 20,
    fontFamily: "inherit",
    transition: "background-color 0.3s ease-out",
    maxWidth: theme.spacing(1) * 60,
    padding: theme.spacing(2, 3),
    backgroundColor: "rgba(255,255,255,0)",
    [theme.breakpoints.only("xs")]: {
      padding: theme.spacing(1.5, 2)
    },
    [theme.breakpoints.up("lg")]: {
      width: theme.spacing(1) * 60
    },
    "&:not(:last-of-type)": {
      marginBottom: theme.spacing(1)
    },
    "&::before": {
      content: "''",
      position: "absolute",
      left: 0,
      top: 0,
      width: "100%",
      height: "100%",
      backgroundColor: "currentColor",
      transition: "opacity 0.2s ease-out",
      opacity: 0.1
    },
    "&:focus, &:hover": {
      "&::before": {
        opacity: 0.05
      }
    }
  },
  key: {
    opacity: 0.5,
    width: "1em",
    wordBreak: "keep-all",
    textAlign: "center",
    flexShrink: 1,
    marginRight: theme.spacing(3)
  },
  selected: {
    color: theme.palette.primary.contrastText,
    "&::before": {
      backgroundColor: theme.palette.primary.main,
      opacity: 1
    },
    "&:focus, &:hover": {
      "&::before": {
        opacity: 1
      }
    }
  },
  text: {
    color: "currentColor",
    zIndex: 1,
    wordBreak: "break-word"
  }
}));

const ResponseButton = ({
  children,
  selected = false,
  handleClick,
  responseKey = "",
  ...props
}) => {
  const classes = useStyles();
  return (
    <ButtonBase
      className={classNames(classes.root, selected && classes.selected)}
      onClick={handleClick}
      {...props}
    >
      <Box className={classes.key}>{responseKey}</Box>
      <Box className={classes.text}>{children}</Box>
    </ButtonBase>
  );
};
export default ResponseButton;
