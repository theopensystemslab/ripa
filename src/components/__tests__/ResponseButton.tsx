import { cleanup, fireEvent, render } from "@testing-library/react";
import * as React from "react";

import ResponseButton from "../ResponseButton";

afterEach(cleanup);

describe("Response Button Component", () => {
  let handleClick;
  let response;
  let responseKey;
  beforeEach(() => {
    handleClick = jest.fn();
    responseKey = "A";
    response = {
      id: "1",
      text: "inner text"
    };
  });
  it("should render snapshot when selected is true", () => {
    const { asFragment } = render(
      <ResponseButton
        key={response.id}
        handleClick={handleClick}
        selected={true}
        responseKey={responseKey}
      >
        {response.text}
      </ResponseButton>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render snapshot when selected is false", () => {
    const { asFragment } = render(
      <ResponseButton
        key={response.id}
        handleClick={handleClick}
        selected={false}
        responseKey={responseKey}
      >
        {response.text}
      </ResponseButton>
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it("should trigger handleClick when clicked", () => {
    const { getByText } = render(
      <ResponseButton
        key={response.id}
        handleClick={handleClick}
        selected={false}
        responseKey={responseKey}
      >
        {response.text}
      </ResponseButton>
    );
    fireEvent.click(getByText("inner text"));
    expect(handleClick).toHaveBeenCalled();
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
