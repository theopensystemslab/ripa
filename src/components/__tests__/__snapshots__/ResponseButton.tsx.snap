// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response Button Component should render snapshot when selected is false 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root makeStyles-root-51"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiBox-root MuiBox-root-59 makeStyles-key-52"
    >
      <span
        class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-alignLeft"
      >
        A
      </span>
    </div>
    <div
      class="MuiBox-root MuiBox-root-90 makeStyles-text-54"
    >
      <span
        class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-alignLeft"
      >
        inner text
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`Response Button Component should render snapshot when selected is true 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root makeStyles-root-1 makeStyles-selected-3"
    tabindex="0"
    type="button"
  >
    <div
      class="MuiBox-root MuiBox-root-9 makeStyles-key-2"
    >
      <span
        class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-alignLeft"
      >
        A
      </span>
    </div>
    <div
      class="MuiBox-root MuiBox-root-40 makeStyles-text-4"
    >
      <span
        class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit MuiTypography-alignLeft"
      >
        inner text
      </span>
    </div>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
